audiostream/src/ni/media/audio/sink/aiff_sink.h

```diff
@@ -0,0 +1,191 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//
+
+#pragma once
+
+#include <ni/media/audio/aiff/aiff_chunks.h>
+#include <ni/media/audio/aiff/aiff_ofstream_info.h>
+#include <ni/media/audio/ieee80.h>
+#include <ni/media/iostreams/device/subview.h>
+#include <ni/media/iostreams/write_obj.h>
+
+//----------------------------------------------------------------------------------------------------------------------
+
+namespace detail
+{
+
+template <class Sink>
+auto write_aiff_header( Sink& sink )
+{
+    // FORM tag
+    write_obj( sink, boost::endian::big_uint32_t( aiff::tags::form ) );
+    sink.file_size_offset( sink.tell() );
+    write_obj( sink, boost::endian::big_uint32_t( 0 ) );
+    write_obj( sink, boost::endian::big_uint32_t( aiff::tags::aifc ) );
+
+    // COMM
+    write_obj( sink, boost::endian::big_uint32_t( aiff::tags::comm ) );
+    write_obj( sink, boost::endian::big_uint32_t( 18 ) );
+    write_obj( sink, boost::endian::big_uint16_t( sink.info().num_channels() ) );
+    sink.num_frames_offset( sink.tell() );
+    write_obj( sink, boost::endian::big_uint32_t( 0 ) );
+    write_obj( sink, boost::endian::big_uint16_t( sink.info().format().bitwidth() ) );
+    uint8_t extSampleRateBuffer[10];
+    double_to_ieee_80( static_cast<double>( sink.info().sample_rate() ), extSampleRateBuffer );
+    write_obj( sink, extSampleRateBuffer );
+
+    // compression
+    if ( sink.info().format().endian() == pcm::endian_type::little_endian )
+    {
+        write_obj( sink, boost::endian::big_uint32_t( aiff::tags::sowt ) );
+    }
+    else if ( sink.info().format().number() == pcm::number_type::floating_point )
+    {
+        if ( sink.info().format().bitwidth() == pcm::bitwidth_type::_32bit )
+        {
+            write_obj( sink, boost::endian::big_uint32_t( aiff::tags::fl32 ) );
+        }
+        else if ( sink.info().format().bitwidth() == pcm::bitwidth_type::_64bit )
+        {
+            write_obj( sink, boost::endian::big_uint32_t( aiff::tags::fl64 ) );
+        }
+        else
+        {
+            throw std::runtime_error( "Unknown/unsupported floating point bitwidth" );
+        }
+    }
+    else
+    {
+        write_obj( sink, boost::endian::big_uint32_t( aiff::tags::none ) );
+    }
+
+    // SSND
+    write_obj( sink, boost::endian::big_uint32_t( aiff::tags::ssnd ) );
+    sink.ssnd_size_offset( sink.tell() );
+    write_obj( sink, boost::endian::big_uint32_t( 0 ) );
+    write_obj( sink, aiff::SoundDataChunk{0, 0} );
+
+    sink.header_size( sink.tell() );
+}
+
+template <class Sink>
+auto close_aiff( Sink& sink )
+{
+    uint32_t file_size = static_cast<uint32_t>( sink.tell() );
+
+    sink.seek( sink.file_size_offset(), BOOST_IOS::beg );
+    write_obj( sink, boost::endian::big_uint32_t( file_size - 8 ) );
```

Comment:
Where is the 8 comming from? Is this the header size? In this case please use `sink.header_size()`

Reply:
the 8 represents the 4 bytes before it ("FORM") and the 4 bytes that stores the file size minus those 8 bytes.

the wording is confusing but this should help explain it: http://www.onicos.com/staff/iz/formats/aiff.html